### PR TITLE
Form validation

### LINKED ISSUE #97: Implement validators for Forms
# Plan/Proposed Change
Need to have validators in place for required fields.

Might want to use validator.js package to help with this.

### LINKED ISSUE #137: Handle null values in front-end
# Plan/Proposed Change
Front end needs to handle null values (i.e., posting with no picture, posting with no category, etc)

### PR DESCRIPTION
**Pull request by EHaake**
## Additions
Finally we have form validation!!
Utilized Formik and Yup to implement form validation on:
1) Login Screen
2) Register Screen
3) Profile Creation Screen
4) Profile Edit Screen
5) Posting Creation Screen

Mostly basic string validation with required fields. Form will not submit if required fields are not filled out. Email addresses are validated on the front end.  This should remove most cases where things could screw up on the back end. 

## Other Changes
1) Also modified to styling slightly so that error messages per input can be displayed without wrecking the layout.
2) Adding a profile image on Profile Creation is no longer required.
3) If user doesn't fill out the Profile Text field, then generic text "This is my profile" is added form them since currently the back end required profile text to be submitted.
4) Adding an image to a posting on Posting Creation Screen is required.

## To Note
Validation not added to Edit Posting Screen since that screen is still under construction.

## Platforms Tested On
Tested and working on iOS (iPhone 6s Plus) and Android Emulator.
Tested  and working with both Ace-mote and cellular-virtue back-ends.

## Closed Issues
closes #97 
closes #137

### CODE REVIEW

File: react-native/providers/AuthProvider.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export const AuthProvider = props => {
     const fetchCommunities = async () => {
         await AsyncStorage.getItem('communities').then(communityData => {
             console.log('Fetching communties from AsyncStorage: ');
-            if (communityData) {
+            if (false) {
```
Comment (SimonBartonPSU) [inline react-native/providers/AuthProvider.js:108]: This won't execute. Is this intentional for now?

**Review by SimonBartonPSU — COMMENTED** [review]
Yup, awesome work. This was no small task. One note:

Profile creation seems to not let me update my community in the action sheet on iOS (just stays Goose Hollow no matter what).

Same thing with Profile Edit screen.

Otherwise looks awesome. :+1:  I like the look of the little red text. I will test again and on Droid later this evening.

File: react-native/providers/AuthProvider.js (hunk 1 above)
Comment (EHaake) [reply react-native/providers/AuthProvider.js:108]: Oh yeah. Good catch. Was just trying always fetching communities. Not meant to be a permanent change.

**Comment by EHaake** [comment]
Yeah, the iOS issue is being addressed in #223. Allie was experimenting with different picker types and looks like we settled on one that will be used project wide so we won't have to do completely separate components for both iOS and android.

**Pull request merged**